src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -31,6 +33,12 @@ public IEnumerator<TElement> GetEnumerator()
                 {
                     yield return buffer._items[map[i]];
                 }
+
+                var rentedMap = Interlocked.Exchange(ref map, null);
```

Comment:
This isn't the cheapest operation. It wouldn't surprise me if this cost more than the use of buffers saved, especially with small inputs.